nimsuggest/tests/tdot4.nim

```diff
@@ -1,9 +1,8 @@
 discard """
-disabled:true
 $nimsuggest --tester --maxresults:2 $file
 >sug $1
 sug;;skProc;;tdot4.main;;proc (inp: string): string;;$file;;10;;5;;"";;100;;None
-sug;;skProc;;strutils.replace;;proc (s: string, sub: string, by: string): string{.noSideEffect, gcsafe, locks: 0.};;$lib/pure/strutils.nim;;1506;;5;;"Replaces `sub` in `s` by the string `by`.";;100;;None
+sug;;skFunc;;strutils.replace;;proc (s: string, sub: string, by: string): string{.noSideEffect, gcsafe, locks: 0.};;*/lib/pure/strutils.nim;;*;;5;;"Replaces `sub` in `s` by the string `by`.*";;100;;None
```

Comment:
please add dedicated module(s) only imported by nimsuggest tests:
`nimsuggest/tests/mimported.nim`
otherwise we'll end up with nimsuggest tests always failing everytime a change is made to those modules in stdlib.

(as you've experienced in your PR)

nimsuggest tests should only rely on its own support files `nimsuggest/tests/m*.nim`

Reply:
Awesome, thanks for the detailed feedback, @timotheecour.

I was wondering about how to isolate for those and my head jumped to using metamorphic testing, but it's likely far simpler to do it this wya. The only issue is that I believe no matter what `system.nim` will always get in there. That might also be a good thing.

Reply:
Additionally, should `dep_v*.nim` become `mdep_v*.nim` -- maybe not in this PR but eventually?

I tried looking up the convention, I do remember reading about `m*` prefix, but I can't remember the precise convention and I'm struggling to find it again. :(

Reply:
> Additionally, should dep_v*.nim become mdep_v*.nim

sure, noone will blame you if you do it in this PR :)

>  I can't remember the precise convention and I'm struggling to find it again. :(

https://nim-lang.github.io/Nim/contributing.html 

> Each test has its own file. All test files are prefixed with t. If you want to create a file for import into another test only, use the prefix m.

(slightly different context but same spirit)

> no matter what system.nim will always get in there.

`from system import nil` as a 1st statement is an option.
```nim
from system import nil
# or:
from system import int, string ...
```

another option is filtering out "flaky things" that include line numbers from system.nim via `execShellCmdEx` based tests as shown above (see trunner.nim which relies on this pattern)

Reply:
Things thus far:
* `deps_v*.nim` are `mdeps_v*.nim`
  * this also stopped them from being compiled needlessly and executables being left behind
* starting to create `m*.nim` files to isolate tests, tests are faster, more stable, and scenarios more precise
  * need to finish moving the rest over
  * `m*.nim` files might grow a fair bit, hopefully it's not a huge deal
* `from system import ...` is a slick trick, for some reason it didn't occur to me that it would work with system 😅 

Thanks again for your support -- it's really welcoming.

Reply:
you're welcome: )

> from system import ... is a slick trick, for some reason it didn't occur to me that it would work with system 😅

PR welcome to document it in https://nim-lang.github.io/Nim/system.html ; searching for `from system` in there returns nothing

Reply:
Sadly, at least for nimsuggest this no longer works, I'm suspecting it was introduced as part of #15935.

Now what happens is that I get a whole lot of system.* suggestions immediatelly following the mstrutils.replace one. I haven't understood the IC stuff well enough to know what exactly is happening and more importantly whether this is in fact invalid.

As an aside I was poking as the VS Code extension to see if I could improve the integration with nimsuggest and this might also explain the increase in crashes. I'll have a further look tomorrow to see if I can suss out what's happening.

Output from the test failure -- I had tester.nim output more than 200 characters:
```
==== STDIN ======================================

Test failed: /home/saem/Development/nim/Nim/nimsuggest/tests/tdot4.nim
  Expected:  *mstrutils.nim     10      5       "this is a test version of strutils.replace, it simply returns `by`"    100     None
sug     skFunc  mstrutils.rereplace     proc (s: string, sub: string, by: string): string{.noSideEffect, gcsafe,
  But got:   /home/saem/Development/nim/Nim/nimsuggest/tests/mstrutils.nim      5       5       "this is a test version of strutils.replace, it simply returns `by`"    100     None
sug     skProc  system.add      proc (x: var string, y: string){.noSideEffect.} /home/saem/Development/nim/Nim/lib/system.nim   1026    5       "Concatenates `x` and `y` in place.\x0A\x0A.. code-block:: Nim\x0A  var tmp = \"\"\x0A  tmp.add(\"ab\")\x0A  tmp.add(\"cd\")\x0A  assert(tmp == \"abcd\")"  100     None
sug     skProc  system.len      proc (x: string): int{.noSideEffect.}   /home/saem/Development/nim/Nim/lib/system.nim   697     5       "Returns the length of a string.\x0A\x0A.. code-block:: Nim\x0A  var str = \"Hello world!\"\x0A  echo len(str) # => 12"     100     None
```

Reply:
Will address this in a future PR after I learn some more about what should happen.